metricflow-semantics/metricflow_semantics/model/semantics/metric_lookup.py

```diff
@@ -135,16 +135,20 @@ def linkable_elements_for_no_metrics_query(
         return result
 
     def linkable_elements_for_metrics(
-        self, metric_references: Sequence[MetricReference], element_set_filter: GroupByItemSetFilter
+        self, metric_references: Sequence[MetricReference], set_filter: GroupByItemSetFilter
     ) -> BaseGroupByItemSet:
         """Retrieve the matching set of linkable elements common to all metrics requested (intersection)."""
-        cache_key = (metric_references, element_set_filter)
+        cache_key = (metric_references, set_filter)
         result = self._group_by_items_for_metrics_cache.get(cache_key)
         if result is not None:
             return result
 
-        result = self._group_by_item_set_resolver.get_linkable_elements_for_metrics(
-            metric_references=metric_references, element_filter=element_set_filter
+        # This method has never returned group-by metrics, so ensure via filter.
```

Comment:
Note about this in case you didn't already have this context - previously, `linkable_elements_for_metrics` was only used in the APIs for listing metadata. We keep group by metrics out of the list since that can be an extremely long list of options, and it's also not supported in the group by (only filters).
I don't think you did this anywhere, but in case I overlook anything, just want to make sure we're not using this method to resolve supported group bys anywhere that group by metrics actually should be supported.

Reply:
Noted, thanks!